### DISCUSSION TITLE
Using ZenFS with Webpack?

### DISCUSSION THREAD

**Discussion by wch** [zen-fs/core#111] [Q&A] [answered]
I tried using BrowserFS for a project, but it didn't work because its implementation of `readdirSync()` does not support `options`. Specifically, I need support for `{ withFileTypes: true }` so that it will return `Array<Dirent>`.

It looks like ZenFS _does_ support `withFileTypes`, which is great.

However, I haven't been able to figure out how to use ZenFS with Webpack. The BrowserFS readme file has very detailed instructions on how to configure it to work with Webpack, but the ZenFS readme doesn't contain this information. Could you provide instructions on how to use it with Webpack?

Much of the code I'm using uses `fs`, but I don't control that code -- it's brought in by other modules, so I am hoping that Webpack will be able to shim in `fs`. (I believe I also need shims for `buffer` and `path`, and possibly `process` as well.)

**Comment by james-pre** [comment, marked as answer]
ZenFS has a named export `fs`, which is also the default export. This export is a drop-in replacement for Node's `fs` module. ZenFS does *not* require you to shim `buffer`, `path`, or `process`.

For usage with Webpack, you should be able to use something like this in your config:
```json
{
	"resolve": {
		"alias": {
			"fs": "@zenfs/core"
		}
	}
}
```

I apologize for not getting back to you sooner, I did not have the correct notification settings.
